src/h5web/explorer/Explorer.tsx

```diff
@@ -19,9 +19,11 @@ interface Props {
 
 function Explorer(props: Props): ReactElement {
   const { onSelect, selectedEntity } = props;
-
   const { domain, metadata } = useContext(ProviderContext);
-  const root = useMemo(() => buildTree(metadata, domain), [domain, metadata]);
+
+  const root = useMemo(() => {
+    return isMetadataMyGroup(metadata) ? metadata : buildTree(metadata, domain);
```

Comment:
I'm temporarily changing the type of `metadata` in the context to `HDF5Metadata | MyHDF5Group` so we can refactor the providers one by one. As a result, I need to use a temporary guard here to know